agent-install/agent-install.sh

```diff
@@ -1821,12 +1951,11 @@ function install_cluster() {
     # create cluster namespace and resources
     create_cluster_resources
 
-    while [[ -z ${RESOURCE_READY} && ${GET_RESOURCE_MAX_TRY} -gt 0 ]]; do
-        #lily: since the purpose of check_resources_for_deployment() is to return 0 or 1, better to have the function actually return that and they run it in the while statemetn above
-        check_resources_for_deployment
+    while ! check_resources_for_deployment && [[ ${GET_RESOURCE_MAX_TRY} -gt 0 ]]; do
         count=$((GET_RESOURCE_MAX_TRY - 1))
         GET_RESOURCE_MAX_TRY=$count
     done
+    #lily: shouldn't we handle the case where GET_RESOURCE_MAX_TRY reached 0 but the resources still weren't ready?
```

Comment:
ah yes, you are right. I think it should return error message saying that resource is not ready for agent deployment